spec/sec-ecmascript-function-objects-patch.html

```diff
@@ -0,0 +1,155 @@
+        </th>
+      </tr>
+      <tr>
+        <td>
+          [[Environment]]
+        </td>
+        <td>
+          Environment Record
+        </td>
+        <td>
+          The Environment Record that the function was closed over. Used as the outer environment when evaluating the code of the function.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          [[FormalParameters]]
+        </td>
+        <td>
+          Parse Node
+        </td>
+        <td>
+          The root parse node of the source text that defines the function's formal parameter list.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          [[ECMAScriptCode]]
+        </td>
+        <td>
+          Parse Node
+        </td>
+        <td>
+          The root parse node of the source text that defines the function's body.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          [[ConstructorKind]]
+        </td>
+        <td>
+          ~base~ | ~derived~
+        </td>
+        <td>
+          Whether or not the function is a derived class constructor.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          [[Realm]]
+        </td>
+        <td>
+          Realm Record
+        </td>
+        <td>
+          The realm in which the function was created and which provides any intrinsic objects that are accessed when evaluating the function.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          [[ScriptOrModule]]
+        </td>
+        <td>
+          Script Record or Module Record
+        </td>
+        <td>
+          The script or module in which the function was created.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          [[ThisMode]]
+        </td>
+        <td>
+          ~lexical~ | ~strict~ | ~global~
+        </td>
+        <td>
+          Defines how `this` references are interpreted within the formal parameters and code body of the function. ~lexical~ means that `this` refers to the *this* value of a lexically enclosing function. ~strict~ means that the *this* value is used exactly as provided by an invocation of the function. ~global~ means that a *this* value of *undefined* or *null* is interpreted as a reference to the global object, and any other *this* value is first passed to ToObject.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          [[Strict]]
+        </td>
+        <td>
+          Boolean
+        </td>
+        <td>
+          *true* if this is a strict function, *false* if this is a non-strict function.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          [[HomeObject]]
+        </td>
+        <td>
+          Object
+        </td>
+        <td>
+          If the function uses `super`, this is the object whose [[GetPrototypeOf]] provides the object where `super` property lookups begin.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          <ins>[[ClassObject]]</ins>
+        </td>
+        <td>
+          <ins>Object | *undefined*</ins>
```

Comment:
Should this be "Function | undefined", since a non-constructible object should never be allowed here?

Reply:
Possibly.

Reply:
Note, however, that `[[FunctionObject]]` above it is listed as `Object` in that table.

Reply:
True enough. still might be useful to have the description describe it as a function?